plugins/modules/api.py

```diff
@@ -334,38 +334,24 @@ def __init__(self):
         self.update = self.module.params['update']
         self.arbitrary = self.module.params['cmd']
 
-        self.where = None
-        self.query = self.module.params['query']
-        if self.query:
-            where_index = self.query.find(' WHERE ')
-            if where_index < 0:
-                self.query = self.split_params(self.query)
-            else:
-                where = self.query[where_index + len(' WHERE '):]
-                self.query = self.split_params(self.query[:where_index])
-                # where must be of the format '<attribute> <operator> <value>'
-                m = re.match(r'^\s*([^ ]+)\s+([^ ]+)\s+(.*)$', where)
-                if not m:
-                    self.errors("invalid syntax for 'WHERE %s'" % where)
-                try:
-                    self.where = [
-                        m.group(1),  # attribute
-                        m.group(2),  # operator
-                        parse_argument_value(m.group(3).rstrip())[0],  # value
-                    ]
-                except ParseError as exc:
-                    self.errors("invalid syntax for 'WHERE %s': %s" % (where, exc))
-            try:
-                idx = self.query.index('WHERE')
-                self.where = self.query[idx + 1:]
-                self.query = self.query[:idx]
-            except ValueError:
-                # Raised when WHERE has not been found
-                pass
-
         self.result = dict(
             message=[])
 
+        self.query = self.module.params['query']
+        if self.query:
+            if "items" not in self.query.keys():
```

Comment:
You should let Ansible do (some of) these checks (by using `options=dict(...)` in the argument spec for the new option).